[improve][doc] Add note for using non-persistent topic in TableView (#18901)
apache/pulsar: site2/docs/client-libraries-java.md

:::note

Each TableView uses one Reader instance per partition, and reads the topic starting from the compacted view by default. It is highly recommended to enable automatic compaction by [configuring the topic compaction policies](cookbooks-compaction.md#configure-compaction-to-run-automatically) for the given topic or namespace. More frequent compaction results in shorter startup times because less data is replayed to reconstruct the TableView of the topic.

> TableView also supports reading from the non-persistent topic since version 2.11.0, but it does not guarantee data consistency.

Reviewer: ```suggestion
Starting from Pulsar 2.11.0, TableView also supports reading non-persistent topics, but it does not guarantee data consistency.
```
